internal/injector/builtin/yaml/databases/redigo.yml

```diff
@@ -16,23 +16,71 @@ aspects:
     join-point:
       function-call: github.com/gomodule/redigo/redis.Dial
     advice:
-      - replace-function:
-          gopkg.in/DataDog/dd-trace-go.v1/contrib/gomodule/redigo.Dial
+      - wrap-expression:
+          imports:
+            redigo: github.com/gomodule/redigo/redis
+            redigotrace: gopkg.in/DataDog/dd-trace-go.v1/contrib/gomodule/redigo
+          template: |-
+            func() (redis.Conn, error) {
+              {{ $len := len .AST.Args }}
+              {{ if le $len 2 }}
+                return redigotrace.Dial({{ index .AST.Args 0 }}, {{ index .AST.Args 1 }})
+              {{ else }}
+                opts := {{ index .AST.Args 2 }}
```

Comment:
We should unit test these aspect with all possible call styles, because I suspect this isn't gonna work properly in some... This is a variadic function, so it can be:

- `redigo.Dial(x)` (OK)
- `redigo.Dial(x, y)` (Suspect not ok)
- `redigo.Dial(x, y, z, a, b, c)` (Suspect not ok)
- `redigo.Dial(x, arr...)` (OK)

Reply:
There was indeed an issue. I fixed it @RomainMuller